cli/endpoints/batch/deploy-models/heart-classifier-mlflow/deployment-custom/code/batch_driver.py

```diff
@@ -19,18 +20,23 @@ def init():
 
     # Load the model, it's input types and output names
     model = mlflow.pyfunc.load(model_path)
-    if model.metadata.signature.inputs:
-        model_input_types = dict(
-            zip(
-                model.metadata.signature.inputs.input_names(),
-                model.metadata.signature.inputs.pandas_types(),
+    if model.metadata:
+        if model.metadata.signature.inputs:
+            model_input_types = dict(
+                zip(
+                    model.metadata.signature.inputs.input_names(),
+                    model.metadata.signature.inputs.pandas_types(),
+                )
             )
+        if model.metadata.signature.outputs:
+            if model.metadata.signature.outputs.has_input_names():
+                model_output_names = model.metadata.signature.outputs.input_names()
+            elif len(model.metadata.signature.outputs.input_names()) == 1:
+                model_output_names = ["prediction"]
+    else:
```

Comment:
Same comment as below. Shouldn't we log this warning in the else part of 'model.metadata.signature.inputs'?

Reply:
Updated!